source/extensions/quic_listeners/quiche/platform/quic_thread_impl.h

```diff
@@ -48,6 +49,10 @@ class QuicThreadImpl {
     thread_ = nullptr;
   }
 
+  void setThreadFactory(Envoy::Thread::ThreadFactory& thread_factory) {
```

Comment:
Just curious if it's possible to require this as a ctor arg rather than having a ctor.

Reply:
Unfortunately no, because `QuicThreadImpl` is a dependency of an [external dependency](https://quiche.googlesource.com/quiche/+/master/quic/platform/api/quic_thread.h) that is subclassing and calling the single arg constructor.

Reply:
cool; might want to mention that in a comment here in case there's confusion about this pattern, but it's up to you.